tracing-error/README.md

```diff
@@ -88,8 +87,67 @@ impl MyError {
 }
 ```
 
-In the future, this crate may also provide its own `Error` types as well,
-for users who do not wish to use other error-handling libraries.
+This crate also provides the [`InstrumentResult`] and [`InstrumentError`]
+traits, which can be used to wrap errors with a [`TracedError`] which
+bundles the inner error with a [`SpanTrace`].
```

Comment:
nit/take it or leave it: i might introduce `TracedError` first, before talking about the `Instrument$TY` traits? not a blocker